### PR TITLE
Update Avocado package

### PR DESCRIPTION
**Pull request by JL2210**
Tested on x86_64

### CODE REVIEW

File: packages/avocado.rb [hunk 1]
```diff
@@ -3,36 +3,29 @@
...
   })
   binary_sha256 ({
-    aarch64: 'f434f34711084c9e425b6c6ea9f6a3412274cb41b2ab59027bc3919c811a2c7a',
-     armv7l: 'f434f34711084c9e425b6c6ea9f6a3412274cb41b2ab59027bc3919c811a2c7a',
-       i686: 'f8284b6aa7b57571759724904979a0052ff42abd0009b2b785c03440ad365437',
-     x86_64: 'c8db96fe48bdade9f974931f6eac557f7e2d65292a335449b4fdbc6c47b4ee53',
   })
 
-  depends_on 'python27' unless File.exists? "#{CREW_PREFIX}/bin/pip"
+  depends_on 'python3'
```
[the remark this answers is not in the document]
Comment (JL2210) [reply packages/avocado.rb]: It's newer. I figured it's better to switch mow than when Python 2 is deprecated.

File: packages/avocado.rb [hunk 2]
```diff
@@ -3,36 +3,29 @@
...
   depends_on 'xzutils'
+  depends_on 'setuptools'
+
+  def self.build
+    system "python3 setup.py build"
+  end
 
   def self.install
-    system "pip install avocado-framework -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
+    system "pip2 install avocado-framework -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
```
[the remark this answers is not in the document]
Comment (JL2210) [reply packages/avocado.rb]: Yeah.

File: packages/avocado.rb [hunk 3]
```diff
@@ -3,36 +3,29 @@
...
-    aarch64: 'f434f34711084c9e425b6c6ea9f6a3412274cb41b2ab59027bc3919c811a2c7a',
-     armv7l: 'f434f34711084c9e425b6c6ea9f6a3412274cb41b2ab59027bc3919c811a2c7a',
-       i686: 'f8284b6aa7b57571759724904979a0052ff42abd0009b2b785c03440ad365437',
-     x86_64: 'c8db96fe48bdade9f974931f6eac557f7e2d65292a335449b4fdbc6c47b4ee53',
   })
 
-  depends_on 'python27' unless File.exists? "#{CREW_PREFIX}/bin/pip"
+  depends_on 'python3'
   depends_on 'xzutils'
+  depends_on 'setuptools'
```
Comment (uberhacker) [inline packages/avocado.rb]: Probably should add `depends_on 'xdg_base'` to accommodate `$HOME/.config/avocado/avocado.conf`.

File: packages/avocado.rb [hunk 4]
```diff
@@ -3,36 +3,29 @@
...
+
+  def self.build
+    system "python3 setup.py build"
+  end
 
   def self.install
-    system "pip install avocado-framework -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
+    system "pip3 install avocado-framework -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
+    system "python3 setup.py install --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
```
Comment (uberhacker) [inline packages/avocado.rb]: You don't need the self.build section or this line if you use pip3 to install.

Comment (uberhacker) [reply packages/avocado.rb]: This should lock down the version:
```
system "pip3 install avocado-framework==64.0 -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
```

File: packages/avocado.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (uberhacker) [reply packages/avocado.rb]: This line should be removed if using `depends_on 'setuptools'`.

[the remark this answers is not in the document]
Comment (JL2210) [reply packages/avocado.rb]: True.

File: packages/avocado.rb (hunk 3 above)
Comment (JL2210) [reply packages/avocado.rb]: OK.

File: packages/avocado.rb (hunk 4 above)
Comment (JL2210) [reply packages/avocado.rb]: This makes sure the avocado script is chmod 755. Why download the sources if you don't use anything but requirements.txt?

Comment (uberhacker) [reply packages/avocado.rb]: Exactly, why use `python3 setup.py install` if you are already installing via `pip3 install`?

Comment (JL2210) [reply packages/avocado.rb]: So I can change the source URL to file:///dev/null and change the sha256sum to the sha256sum of an empty file?

Comment (JL2210) [reply packages/avocado.rb]: I'm simply pulling in the dependencies with pip.

Comment (uberhacker) [reply packages/avocado.rb]: Remove the self.build section and change the line below and I'll merge it:
```
system "pip3 install -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
```

Comment (JL2210) [reply packages/avocado.rb]: Why? Isn't it better to build from source, considering that the executable bit isn't set with the pip package?

Comment (JL2210) [reply packages/avocado.rb]: Fair enough. I'll give you write access.

**Comment by JL2210** [comment]
How's this now?

File: packages/avocado.rb [hunk 5]
```diff
@@ -3,36 +3,25 @@
...
   })
 
-  depends_on 'python27' unless File.exists? "#{CREW_PREFIX}/bin/pip"
+  depends_on 'xdg_base'
   depends_on 'xzutils'
+  depends_on 'setuptools'
 
   def self.install
-    system "pip install avocado-framework -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
+    system "pip3 install avocado-framework==64.0 -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
```
Comment (uberhacker) [inline packages/avocado.rb]: Take out the avocado-framework argument:
```
system "pip3 install -r requirements.txt --root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
```

Comment (JL2210) [reply packages/avocado.rb]: OK. I understand now.

Comment (JL2210) [reply packages/avocado.rb]: What I don't understand is why we couldn't have done that with AWS.

**Comment by uberhacker** [comment]
You are right.  We could have done it this way with aws also.  The point I've been trying to drive home is if you want to use pip3 to install the package, then use pip3.  Otherwise, just use pip3 to install the requirements ONLY and then the package via python3 setup.py install.

**Comment by JL2210** [comment]
Yeah. Thank you for explaining. Ready?

**Pull request merged**